lib/iris/analysis/maths.py

```diff
@@ -583,11 +579,14 @@ def _binary_op_common(operation_function, operation_name, cube, other,
         other = _broadcast_cube_coord_data(cube, other, operation_name, dim)
     elif isinstance(other, iris.cube.Cube):
         try:
-            da.broadcast_to(cube.lazy_data(), other.lazy_data())
+            da.broadcast_to(cube.lazy_data(), other.lazy_data().shape)
         except ValueError:
             other = iris.util.as_compatible_shape(other, cube).lazy_data()
         else:
-            other = other.lazy_data()
+            if cube.has_lazy_data() and not in_place:
```

Comment:
I dislike having to check for the existence of lazy / non lazy data in multiple places, but in the current implementation .data is used in at least two different places. This if check makes sure that the 'other' data stored in the unary_func is in the same format as as the cube data. I'd much prefer to check once for laziness then explicitly fork the logic.